modal/stub.py

```diff
@@ -642,11 +639,19 @@ def function(
             name=name,
             cloud=cloud,
             webhook_config=webhook_config,
+            _cls=_cls,
         )
 
         self._add_function(function, [*base_mounts, *mounts])
         return function_handle
 
+    @decorator_with_options_unsupported
+    def method(
+        self,
+        raw_f: Optional[Callable[..., Any]] = None,
+    ):  # TODO: return type!
```

Comment:
actually not sure if we can have a return type here, since the methods are "translated" from a `_PartialFunction` to a `_FunctionHandle` in runtime (using `_PartialFunction.__get__`). 

I think the only way to solve this is to have a shared base class that both derive from, and define that as the return type here.

Reply:
Static type checkers tend to support Descriptors nowadays, so you should be able to have `_PartialFunction` as the return type as long as the `__get__` method on the descriptor has `_FunctionHandle` as its return type. I experimented a bit with this a while ago and it seemed to work, at least in VSCode.

(Using `@typing.overload` on `__get__`, VSCode was even able to infer difference in types between `MyClass.my_descriptor` and `MyClass().my_descriptor`!)

Reply:
Oh very interesting. Let me annotate `__get__` on `PartialFunction` then!